scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1321,7 +1323,13 @@ private boolean isClientInControlOfSourceIpAddress(ClientHello clientHello, Reco
 					if (connection == null || !connection.hasEstablishedSession()) {
 						// only new connections or connections without
 						// established sessions
-						return true;
+						int pending = pendingShortResumptionHandshakes.get();
+						int remainingCapacity = connectionStore.remainingCapacity();
+						if (pending < (remainingCapacity / 8)) {
```

Comment:
remaining capacity is wrong, but the basic idea should work :-)

Reply:
You mean you would like that remaining capacity does not count staled connection ?

Reply:
You mean you would like that remaining capacity does not count staled connection ?

Reply:
Yes (or something similar).
I want to show, how to check pending short resumption and use that number to trigger the HELLO_VERIFY_REQUEST, if that number gets to high. But I'm still investigating. So this just a first idea of such a dynamic defence agains spoofing.

Reply:
FMPOV, it would wire up the serial execution, if we pass it into the connection store.
With the move of the flight and the introduction of the timeout, handshakes should not be ongoing, when connections get evicted.